src/v/cloud_storage/segment_meta_cstore.h

```diff
@@ -823,23 +820,23 @@ class segment_meta_column_impl
     std::list<frame_t> _frames;
 };
 
-/// Segment metadata column.
+/// DeltaFor encoded collection of values.
 ///
 /// Contains a list of frames. The frames are not overlapping with each
 /// other. The iterator is used to scan both all frames seamlessly. This
 /// specialization is for xor-delta algorithm. It doesn't allow skipping
 /// frames so all search operations require full scan. Random access by
 /// index can skip frames since indexes are monotonic.
 template<class value_t>
-class segment_meta_column<value_t, details::delta_xor>
-  : public segment_meta_column_impl<
+class deltafor_column<value_t, details::delta_xor>
+  : public deltafor_column_impl<
       value_t,
       details::delta_xor{},
-      segment_meta_column<value_t, details::delta_xor>> {
-    using base_t = segment_meta_column_impl<
+      deltafor_column<value_t, details::delta_xor>> {
+    using base_t = deltafor_column_impl<
       value_t,
       details::delta_xor{},
-      segment_meta_column<value_t, details::delta_xor>>;
+      deltafor_column<value_t, details::delta_xor>>;
```

Comment:
nit: since we're trying to make this a bit more generic, perhaps it's worth calling out the expected limitations that non-segment-cstore users may need to consider, like the fact that `pred_search` will return the first frame that matches the predicate.